### PR TITLE
Workaround blink/chromium grid layout limitation of 1000 rows

### PR DESCRIPTION
**Pull request by dns2utf8**
I made this in case we don't come up with a better solution in time.

See https://github.com/rust-lang/rust/issues/88545 for more details.

A rendered version of the standard library is hosted here:
https://data.estada.ch/rustdoc-nightly_497ee321af_2021-09-09/core/arch/arm/index.html

r? @GuillaumeGomez @jsha

### CODE REVIEW

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Waiting for then CI then I r+ it.

**Comment by mqudsi** [comment]
*cough* tables *cough*. 

But seriously, if you use a css reset to remove all browser styling (are we already doing that?) you can use the table layout in the renderer to create code like our current grid (because we are barely using any of the actual grid features). A little bit of css to the cell classes makes it responsive.

**Comment by dns2utf8** [comment]
You are welcome! 
Btw. I just realised, if there is a 1.55.1 release: This PR should probably be included in that

**Comment by dns2utf8** [comment]
I tested the workaround with IE Edge 92 and Chrome 93 and it works okay.
Safari on Mac and Firefox worked before already and behave the same as the blink based browsers now ☺️

**Comment by GuillaumeGomez** [comment]
Good point, let's nominate it for beta backport.

Do you accept this to be backported to beta @rust-lang/rustdoc?

**Comment by Nemo157** [comment]
Was the issue a beta regression?

**Comment by GuillaumeGomez** [comment]
I think it reached the 1.56 beta version, so yes, it's a regression.

**Comment by Nemo157** [comment]
I was more wondering whether it was not already in stable, looking back at the history I assume https://github.com/rust-lang/rust/pull/85651 introduced the issue? That is on 1.55, but maybe it's worth a beta backport to only have a single version with the problem.

**Comment by GuillaumeGomez** [comment]
Yes it's the one.

**Comment by dns2utf8** [comment]
Unfortunately it has reached stable too: https://doc.rust-lang.org/beta/core/arch/arm/index.html

**Comment by GuillaumeGomez** [comment]
Ah damn... Oh well, at least it'll only be one version with this issue...

**Comment by dns2utf8** [comment]
If there is a 1.55.1 release we could backport the workaround there too

**Comment by dns2utf8** [comment]
Should I open a PR with this commit on the beta branch too?

**Comment by GuillaumeGomez** [comment]
Yes please. :)

**Comment by dns2utf8** [comment]
Done :) Could you move the beta-nominated label to the new PR?

**Comment by jyn514** [comment]
This is not how beta nominations work. Please use the existing process so the release team can keep track of which PRs have been backported and which haven't.

https://forge.rust-lang.org/release/backporting.html?highlight=Beta%20n#beta-backporting-in-rust-langrust

**Comment by dns2utf8** [comment]
Very sorry. I renamed the other PR, but I am not sure If I should be reopened ir or if someone else will cherrypick one of the PRs or me doing nothing?

**Comment by jyn514** [comment]
@dns2utf8 it's ok, that's why there's docs :) if you want to make a backport PR that's fine, but please include all the other beta accepted PRs at the same time, and ask a member of T-release to review, not Guillaume.

**Comment by camelid** [comment]
It's probably best to just wait for T-release to backport this PR with the other beta backports. Someone from T-release goes through and backports beta-accepted PRs periodically.

**Pull request merged**
